### PR TITLE
96Boards: Adapt board configuration guidelines

### PR DESCRIPTION
**Pull request by Mani-Sadhasivam**
This PR modifies the 96Boards board support to adapt to the default board configuration guidelines as per http://docs.zephyrproject.org/porting/board_porting.html#default-board-configuration. Reference is taken from: https://github.com/zephyrproject-rtos/zephyr/tree/master/boards/arm/nucleo_f429zi

Following supported 96Boards are modified:

1. 96b_argonkey
2. 96b_carbon
3. 96b_carbon_nrf51
4. 96b_nitrogen
5. 96b_neonkey

Fixes: #9233 

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>

### CODE REVIEW

**Comment by Mani-Sadhasivam** [comment]
There is no on-board LED connected to 96b_carbon_nrf51 and hence the `blinky` app in CI fails. This is just a slave BT chip.

**Comment by erwango** [comment]
> There is no on-board LED connected to 96b_carbon_nrf51 and hence the blinky app in CI failed. This is just a slave BT chip on 96b_carbon board

`blinky` yaml indicates `depends_on: gpio`, and this is why test is run on that platform.
Either a new tag should be created, either `gpio` should be removed from board yaml file.

**Review by galak — APPROVED** [review]

**Review by erwango — COMMENTED** [review]
2 Questions otherwise looks fine

File: boards/arm/96b_argonkey/Kconfig.defconfig [hunk 1]
```diff
@@ -13,30 +13,36 @@ config BOARD
 if UART_CONSOLE
 
 config UART_STM32_PORT_1
-	default y
+	def_bool y
```
[the remark this answers is not in the document]
Comment (Mani-Sadhasivam) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: I'm not sure either. Looking at some other boards like `reel_board` I thought that we should use `def_bool` for providing boolean choices and `default` for other than booleans.

File: boards/arm/96b_argonkey/Kconfig.defconfig [hunk 2]
```diff
@@ -13,30 +13,36 @@ config BOARD
...
 
 endif # UART_CONSOLE
 
 if I2C
 
 config I2C_1
-	default y
+	def_bool y
 
 config I2C_2
```
[the remark this answers is not in the document]
Comment (Mani-Sadhasivam) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: I don't have the schematics, but looking at the board doc it looks like [3 I2C's](https://docs.zephyrproject.org/1.12.0/boards/arm/96b_argonkey/doc/96b_argonkey.html#i2c) and [2 SPI's](https://docs.zephyrproject.org/1.12.0/boards/arm/96b_argonkey/doc/96b_argonkey.html#spi) are exposed on the board but not sure where.

File: boards/arm/96b_argonkey/Kconfig.defconfig [hunk 3]
```diff
@@ -13,30 +13,36 @@ config BOARD
...
+	def_bool y
 
 endif # I2C
 
 if SPI
 
+config SPI_1
+	def_bool y
+
 config SPI_2
```
[the remark this answers is not in the document]
Comment (Mani-Sadhasivam) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: Please see above reply.

**Review by nashif — APPROVED** [review]

File: boards/arm/96b_argonkey/Kconfig.defconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: Well... if this on connector then that's fine, if you don't know where it is, this is more of a problem.
Don't 96boards all share more or less the same connector type?

[the remark this answers is not in the document]
Comment (Mani-Sadhasivam) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: As per 96Boards IE spec, only 2 I2C's and 1 SPI is exposed on the LS connector. So my hard guess is, 2 I2C's and 1 SPI are exposed on the connector, and the remaining ones are connected to onboard sensors. Still, I'm skeptical about the design.

[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: @avisconti , any idea?

[the remark this answers is not in the document]
Comment (avisconti) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: @erwango 
I2C2 bus is used to read data from environmental/ALS/proximity sensors
I2C1 bus to access the onboard EEPROM (and available on 40-pin connector also)
I2C3 bus to access led controller plus external sensors thru P2 connector

File: boards/arm/96b_argonkey/Kconfig.defconfig (hunk 3 above)
[the remark this answers is not in the document]
Comment (avisconti) [reply boards/arm/96b_argonkey/Kconfig.defconfig]: @erwango 
SPI1 is available on LS connector
SPI2 bus is internal only to access accel/gyro/mag devices

**Pull request merged**